packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -325,8 +325,10 @@ export class ModelType<S extends ModelProperties, T> extends ComplexType<any, an
                     `Cannot define action '${POST_PROCESS_SNAPSHOT}', it should be defined using 'type.postProcessSnapshot(fn)' instead`
                 )
 
+            let action2 = actions[name].bind(self) // bind to self to allow 'this' to always work
```

Comment:
shouldn't this be `actions[name].bind(actions)`?

Reply:
depends on what you want. if you want it to be able to reference other pieces of code beyond the current scope then it should be "this" (although TS will complain though)
if you **don't** want it to work then yes, it should be "this"

which behaviour is the wanted one?